mslib/msui/mscolab.py

```diff
@@ -990,11 +989,14 @@ def handle_update_permission(self, p_id, u_id, access_level):
             # Close mscolab windows based on new access_level and update their buttons
             if self.access_level == "collaborator" or self.access_level == "viewer":
                 self.adminWindowBtn.setEnabled(False)
-                self.versionHistoryBtn.setEnabled(False)
+                if self.access_level == "collaborator":
+                    self.versionHistoryBtn.setEnabled(True)
+                else:
+                    self.versionHistoryBtn.setEnabled(False)
+                    if self.version_window is not None:
+                        self.version_window.close()
                 if self.admin_window is not None:
                     self.admin_window.close()
-                if self.version_window is not None:
-                    self.version_window.close()
             else:
                 self.adminWindowBtn.setEnabled(True)
                 self.versionHistoryBtn.setEnabled(True)
```

Comment:
The chat window button below seems inconsistent.

The buttons are enabled/disabled at different points in the code. This should be refactored to a single location.
I would prefer also more readable code.

perhaps a standardized code block like
```
allow_version_access = self.access_level in ["..."]
self.versionHistoryBtn.setEnabled(allow_version_access)
if not allow_version_access and self.version_window is not None:
    self.version_window.close()
```
for each button/window ?